src/main/java/org/opentripplanner/model/transfer/TransferPointMap.java

```diff
@@ -73,16 +73,17 @@ E computeIfAbsent(TransferPoint point, Supplier<E> creator) {
   }
 
   /**
-   * List all elements witch matches any of the transfer points added to the map.
+   * List all elements which match any of the transfer points added to the map.
    */
   List<E> get(Trip trip, StopLocation stop, int stopPointInPattern) {
+    Station parentStation = stop != null ? stop.getParentStation() : null;
     return Stream
       .of(
         tripMap.get(new TripKey(trip, stopPointInPattern)),
         routeStopMap.get(new RouteStopKey(trip.getRoute(), stop)),
-        routeStationMap.get(new RouteStationKey(trip.getRoute(), stop.getParentStation())),
+        routeStationMap.get(new RouteStationKey(trip.getRoute(), parentStation)),
```

Comment:
Maybe we need more context here, what kind of data is causing the NullPointerException you are trying to solve here?

Reply:
> @dekarl Can you remember why the null check was necessary?

it is triggered when routing, I have not had time to investigate. But when pondering about it it dawned that it may be a workaround for the effects of #4964.

Now that this was merged I'll rebase and see if it is still needed.

Reply:
After rebasing on top of dev-2.x my test case does not trigger the NPE anymore.

@sjthiessen do your test cases work without this commit?

I think the next step is to get this PR merged without this one commit.